### PR TITLE
Add Vue file icon

### PR DESCRIPTION
**Pull request by d1y**
https://github.com/vuejs/art/blob/master/white-on-dark-logo.svg
image

Release Notes:

- Added icon for `.vue` files.

### CODE REVIEW

File: assets/icons/file_icons/file_types.json
Comment (maxdeviant) [inline assets/icons/file_icons/file_types.json:114]: Can we keep this list sorted alphabetically?

**Comment by SomeoneToIgnore** [comment]
icons

This one is notably above text's center line for me, while others are much better.
Can we do something about it?

File: assets/icons/file_icons/file_types.json
Comment (d1y) [reply assets/icons/file_icons/file_types.json]: Okay, I have made the necessary changes

**Comment by d1y** [comment]
@SomeoneToIgnore I resized the svg manually, it's not fun🤣, Is this okay? If so, I will submit the SVG file that I manually adjusted

**Comment by d1y** [comment]
```diff
diff --git a/assets/icons/file_icons/vue.svg b/assets/icons/file_icons/vue.svg
index 57b656dde..897534089 100644
--- a/assets/icons/file_icons/vue.svg
+++ b/assets/icons/file_icons/vue.svg
@@ -1,5 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="197px" height="170px" version="1.1" viewBox="0 0 197 170">
-    <polygon fill="black" points="98.16 102.01 42.1757011 5.09911711 39.23 0 0 0 2.94507611 5.10107823 98.16 170.02 196.32 0 157.06 0"/>
-    <polygon fill="black" points="98.16 29.0371525 81.35 2.009956e-14 75.5 2.009956e-14 -6.81139173e-16 2.009956e-14 2.94 5.1 78.4487095 5.1 98.16 39.26 117.879366 5.1 193.38 5.1 196.325 0 120.82 7.80656356e-15 114.973219 2.009956e-14"/>
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="197px" height="170px" version="1.1" viewBox="0 0 197 197">
+    <g  transform="translate(8,42)">       
+        <polygon fill="black" points="98.16 102.01 42.1757011 5.09911711 39.23 0 0 0 2.94507611 5.10107823 98.16 170.02 196.32 0 157.06 0"/>
+        <polygon fill="black" points="98.16 29.0371525 81.35 2.009956e-14 75.5 2.009956e-14 -6.81139173e-16 2.009956e-14 2.94 5.1 78.4487095 5.1 98.16 39.26 117.879366 5.1 193.38 5.1 196.325 0 120.82 7.80656356e-15 114.973219 2.009956e-14"/>
+    </g>
 </svg>
\ No newline at end of file

```

**Comment by maxdeviant** [comment]
> @SomeoneToIgnore I resized the svg manually, it's not fun🤣, Is this okay? If so, I will submit the SVG file that I manually adjusted 

It looks a little off-center now.

I think part of the problem might be that there isn't an inherent padding in the SVG, like we have for our other icons. The Vue SVG extends to the full boundaries of the container.

**Review by SomeoneToIgnore — COMMENTED** [review]
Yeah, now it also looks slightly below the baseline (great example is that capital V on your screenshot), but I like this way more than before since current deviations are slightly more subtle.

@maxdeviant has a good eye for those, so I would wait for his judgement here; LGTM for me to merge the current version.

**Review by SomeoneToIgnore — APPROVED** [review]
Oh, now I see it was already @maxdeviant who submitted that, sorry for confusion.

I will leave it for ~12 hours (in case any other objections arise) and merge it, feel free to do that before.

**Comment by naiyerasif** [comment]
@d1y Here's an optically centered version with much smaller footprint.

```xml
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 14 14">
  <path d="M10.6 1.9H13L7 12.292 1 1.9h2.4L7 8.135 10.6 1.9Z"/>
  <path d="M8.386 1.9 7 4.3 5.614 1.9H3.4L7 8.135 10.6 1.9H8.386Z" opacity=".5"/>
</svg>
```

At this point, it should be a good idea to lay down some guidelines around icons, for example, 
- `viewBox` should be `0 0 14 14` with 2px padding around icons
- they should be optically centered (rather than geometrically centered)
- the submitted SVG should be optimized through [svgo](https://svgo.dev/)

Summoning @iamnbutler @maxdeviant to weigh in here. Also, maybe we should discuss this in an issue rather than PR.

**Comment by maxdeviant** [comment]
> @d1y Here's an optically centered version with much smaller footprint.
> 
> ```
> <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 14 14">
>   <path d="M7 8.135 3.578 2.212l-.18-.312H1l6 10.392L13 1.9h-2.4L7 8.135Z"/>
>   <path d="M7 3.675 5.972 1.9H1l.18.312h4.615L7 4.3l1.205-2.088h4.615L13 1.9H8.028L7 3.675Z"/>
> </svg>
> ```
> 
> At this point, it should be a good idea to lay down some guidelines around icons, for example,
> 
> * `viewBox` should be `0 0 14 14` with 2px padding around icons
> * they should be optically centered (rather than geometrically centered)
> * the submitted SVG should be optimized through [svgo](https://svgo.dev/)
> 
> Summoning @iamnbutler @maxdeviant to weigh in here. Also, maybe we should discuss this in an issue rather than PR (?).

@naiyerasif Feel free to open a GitHub discussion about a potential guideline/pipeline for SVGs.

**Comment by d1y** [comment]
now, I used the refreshing svg-content provided by @naiyerasif , which can be merged
image

**Comment by maxdeviant** [comment]
> now, I used the refreshing svg-content provided by @naiyerasif , which can be merged image

Thanks! It looks good to me from your screenshot 🙂

**Pull request merged**
